plugins/woocommerce/client/blocks/tests/e2e/utils/frontend/frontend-utils.page.ts

```diff
@@ -41,27 +41,31 @@ export class FrontendUtils {
 				url.includes( 'batch' )
 			);
 		} );
+		// Wait for all XHR requests to finish. Some blocks do another request to the cart endpoint after the above.
+		// eslint-disable-next-line playwright/no-networkidle
+		await this.page.waitForLoadState( 'networkidle' );
```

Comment:
Using `networkidle` is discouraged as it does not produce a stable outcome.

https://playwright.dev/docs/api/class-frame#frame-wait-for-load-state

Reply:
I'm aware its discouraged hence the lint disabling. 

However I noticed when debugging tests there are several API requests raised in quick succession after adding to cart, possibly due to something in product blocks, which cause an additional request to the `/cart` route. It was my intention to wait for these to finish before continuing because this was the only workaround I had any success with. 

Without this in place the redirect to checkout fails (sometimes) after `addToCart` and you are sent to the cart page. I think it redirects too early.

Reply:
@WunderBart The `cart-item` route check should have been enough since that is the request that actually adds something to cart. And you can see in the failed test videos it does add something because it says "1 in cart" in the block. But I couldn't find anything else at fault.

Reply:
I've been debugging it and found a weird race condition that I described in #56177. Basically, the backend does not seem to catch up to the speed the automation is adding the items to the cart, even if we wait for the `cart/items` request to finish. With 2 items added one by one, we often end up with just one of them, but the weird part is that sometimes the second one is overwriting the first one 🤷

Reply:
Could it be the cart syncing we do in product-button? I adjusted this e2e code recently: https://github.com/woocommerce/woocommerce/pull/56125/files/1a133d036c5e1311af139c83f961804265de3e4f#diff-08c85d077ed71853aa7e713841707635d8fe92e71e84b9ae6f1ad8c5d2c16a73R39

We're about to change the endpoints being called to add to cart so it might change again: https://github.com/woocommerce/woocommerce/pull/56117/files#diff-c5b02e6435a239a6d14da3763adaf8956e615ae3f670a52a5c23e0dc0e337aadR78